CMakeLists.txt

```diff
@@ -29,6 +30,45 @@ project(azure-sdk LANGUAGES CXX)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+if(STATIC_CRT AND MSVC)
+  # 1. More about static/shared CRT:
+  # https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160
+  #
+  # 2. STATIC_CRT build flag approach is used/inspired by libcurl
+  # (https://github.com/curl/curl/blob/master/CMakeLists.txt) and some other projects.
+  #
+  # 3. GTest would emit the following warning:
+  # warning LNK4098: defaultlib 'LIBCMTD' conflicts with use of other libs; use /NODEFAULTLIB:library
+  # AddGoogleTest.cmake uses gtest_force_shared_crt
+  # (see https://github.com/google/googletest/blob/master/googletest/README.md),
+  # which respects linker settings that we set below, and our settings below are all in sync.
+  #
+  # 4. Sometimes, the following approach is recommended instead:
+  # +-----------------------------------------------------------------------------------+
+  # | # Use the static runtime libraries when building statically                       |
+  # | # for consistency with vcpkg's arch-windows-static triplets:                      |
+  # | cmake_policy(SET CMP0091 NEW)                                                     |
+  # | # see https://cmake.org/cmake/help/v3.15/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html |
+  # | if (NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)                                       |
+  # |   set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")         |
+  # | endif()                                                                           |
+  # +-----------------------------------------------------------------------------------+
+  # However, it only works when cmake installed is 3.15+;
+  # we have to require a minimum of 3.13.
+  #
+  # 5. We "replace with empty string" (i.e. remove) first, then add, so that '/MT'
+  # will be present (and present once) even if '/MD' was not.
+
+  string(REGEX REPLACE "/MD" "" CMAKE_CXX_FLAG "${CMAKE_CXX_FLAGS}")
```

Comment:
CMake has four configurations using MSVC: https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
To be complete, this needs to cover all four. Might be better to replace `/MDd` in `_DEBUG` with `/MTd` and replace `/MD` with `/MT` in other configurations.

Reply:
Thank you!